Division by zero in UserInputHandlers

Playing around with the entities sample I got a division by zero exception.

```
2021-10-13 13:59:38,605 [5402799]  ERROR - erver.plugin.InteractionServer - Error WebSocketSession[websocket=JettyListenerEventDriver[org.modelix.ui.sm.server.plugin.InteractionServer],behavior=SERVER,connection=WebSocketServerConnection@464cb91c::SocketChannelEndPoint@7f5383f6{/127.0.0.1:57011<->/127.0.0.1:33333,OPEN,fill=-,flush=-,to=26/300000}{io=0/0,kio=0,kro=1}->WebSocketServerConnection@464cb91c[s=ConnectionState@39f93cf1[OPENED],f=org.eclipse.jetty.websocket.common.io.AbstractWebSocketConnection$Flusher@26db94aa[IDLE][queueSize=0,aggregateSize=-1,terminated=null],g=Generator[SERVER,validating,+rsv1],p=Parser@1a7ee30c[ExtensionStack,s=START,c=0,len=6,f=TEXT[len=0,fin=true,rsv=1..,masked=true]]],remote=WebSocketRemoteEndpoint@39d92d06[batching=true],incoming=JettyListenerEventDriver[org.modelix.ui.sm.server.plugin.InteractionServer],outgoing=ExtensionStack[queueSize=0,extensions=[permessage-deflate],incoming=org.eclipse.jetty.websocket.common.extensions.compress.PerMessageDeflateExtension,outgoing=org.eclipse.jetty.websocket.common.extensions.compress.PerMessageDeflateExtension]]
java.lang.RuntimeException: java.lang.ArithmeticException: / by zero
	at com.intellij.openapi.application.impl.LaterInvocator.invokeAndWait(LaterInvocator.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:476)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:481)
	at org.modelix.ui.sm.server.plugin.InteractionServer.onWebSocketText(InteractionServer.java:87)
	at org.eclipse.jetty.websocket.common.events.JettyListenerEventDriver.onTextMessage(JettyListenerEventDriver.java:231)
	at org.eclipse.jetty.websocket.common.message.SimpleTextMessage.messageComplete(SimpleTextMessage.java:69)
	at org.eclipse.jetty.websocket.common.events.AbstractEventDriver.appendMessage(AbstractEventDriver.java:65)
	at org.eclipse.jetty.websocket.common.events.JettyListenerEventDriver.onTextFrame(JettyListenerEventDriver.java:179)
	at org.eclipse.jetty.websocket.common.events.AbstractEventDriver.incomingFrame(AbstractEventDriver.java:150)
	at org.eclipse.jetty.websocket.common.WebSocketSession.incomingFrame(WebSocketSession.java:321)
	at org.eclipse.jetty.websocket.common.extensions.AbstractExtension.nextIncomingFrame(AbstractExtension.java:147)
	at org.eclipse.jetty.websocket.common.extensions.compress.PerMessageDeflateExtension.nextIncomingFrame(PerMessageDeflateExtension.java:112)
	at org.eclipse.jetty.websocket.common.extensions.compress.CompressExtension.forwardIncoming(CompressExtension.java:168)
	at org.eclipse.jetty.websocket.common.extensions.compress.PerMessageDeflateExtension.incomingFrame(PerMessageDeflateExtension.java:92)
	at org.eclipse.jetty.websocket.common.extensions.ExtensionStack.incomingFrame(ExtensionStack.java:202)
	at org.eclipse.jetty.websocket.common.Parser.notifyFrame(Parser.java:226)
	at org.eclipse.jetty.websocket.common.Parser.parseSingleFrame(Parser.java:262)
	at org.eclipse.jetty.websocket.common.io.AbstractWebSocketConnection.readParse(AbstractWebSocketConnection.java:582)
	at org.eclipse.jetty.websocket.common.io.AbstractWebSocketConnection.onFillable(AbstractWebSocketConnection.java:441)
	at org.eclipse.jetty.websocket.common.io.AbstractWebSocketConnection.onFillable(AbstractWebSocketConnection.java:428)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ArithmeticException: / by zero
	at org.modelix.ui.sm.pf.UserInputHandlers_CCMenu$4.execute(UserInputHandlers_CCMenu.java:197)
	at de.q60.mps.polymorphicfunctions.runtime.RecursionProtection.evaluate(RecursionProtection.java:28)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1$1.invoke(DefaultPFContext.java:52)
	at de.q60.mps.polymorphicfunctions.runtime.ContextValue.computeWith(ContextValue.java:52)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1.select(DefaultPFContext.java:50)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1.select(DefaultPFContext.java:48)
	at jetbrains.mps.internal.collections.runtime.ISelector.invoke(ISelector.java:12)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.moveToNext(SelectingSequence.java:60)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.hasNext(SelectingSequence.java:32)
	at jetbrains.mps.internal.collections.runtime.ListSequence.fromIterable(ListSequence.java:122)
	at jetbrains.mps.internal.collections.runtime.AbstractSequence.toListSequence(AbstractSequence.java:215)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext.callFunctions(DefaultPFContext.java:40)
	at de.q60.mps.polymorphicfunctions.runtime.ContextPFContext.callFunctions(ContextPFContext.java:27)
	at de.q60.mps.polymorphicfunctions.runtime.AbstractPFContext.callFunctions(AbstractPFContext.java:15)
	at org.modelix.ui.sm.pf.UserInputHandlers$4$1.invoke(UserInputHandlers.java:172)
	at de.q60.mps.polymorphicfunctions.runtime.ContextValue.runWith(ContextValue.java:43)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$1$1$1.run(InteractionServer.java:164)
	at de.q60.mps.incremental.util.ContextValue.runWith(ContextValue.java:34)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$1$1.invoke(InteractionServer.java:162)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$1$1.invoke(InteractionServer.java:1)
	at org.modelix.model.mpsadapters.mps.MPSArea$1.run(MPSArea.java:98)
	at jetbrains.mps.smodel.ActionDispatcher.dispatch(ActionDispatcher.java:88)
	at jetbrains.mps.smodel.WorkbenchModelAccess.lambda$runReadAction$0(WorkbenchModelAccess.java:88)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:855)
	at jetbrains.mps.smodel.WorkbenchModelAccess.runReadAction(WorkbenchModelAccess.java:85)
	at jetbrains.mps.smodel.ModelAccessBase.runReadAction(ModelAccessBase.java:56)
	at org.modelix.model.mpsadapters.mps.MPSArea.executeRead(MPSArea.java:96)
	at org.modelix.model.area.CompositeArea$executeRead$1$1$1.invoke(CompositeArea.kt:65)
	at org.modelix.model.api.PBranch$computeRead$1.invoke(PBranch.kt:70)
	at org.modelix.model.api.PBranch$computeRead$1.invoke(PBranch.kt:20)
	at org.modelix.model.api.ContextValue.computeWith(ContextValue.kt:38)
	at org.modelix.model.api.PBranch.runRead(PBranch.kt:41)
	at org.modelix.model.api.PBranch.computeRead(PBranch.kt:70)
	at org.modelix.model.area.PArea$executeRead$1.invoke(PArea.kt:58)
	at org.modelix.model.area.ContextArea.offer(ContextArea.kt:37)
	at org.modelix.model.area.PArea.executeRead(PArea.kt:58)
	at org.modelix.model.area.CompositeArea$executeRead$1$1$1.invoke(CompositeArea.kt:65)
	at org.modelix.model.area.CompositeArea$executeRead$1.invoke(CompositeArea.kt:304)
	at org.modelix.model.api.ContextValue.computeWith(ContextValue.kt:38)
	at org.modelix.model.area.ContextArea.offer(ContextArea.kt:35)
	at org.modelix.model.area.CompositeArea.executeRead(CompositeArea.kt:65)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$1.runRead(InteractionServer.java:160)
	at org.modelix.ui.sm.pf.UserInputHandlers$4.execute(UserInputHandlers.java:170)
	at de.q60.mps.polymorphicfunctions.runtime.RecursionProtection.evaluate(RecursionProtection.java:28)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1$1.invoke(DefaultPFContext.java:52)
	at de.q60.mps.polymorphicfunctions.runtime.ContextValue.computeWith(ContextValue.java:52)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1.select(DefaultPFContext.java:50)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext$1.select(DefaultPFContext.java:48)
	at jetbrains.mps.internal.collections.runtime.ISelector.invoke(ISelector.java:12)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.moveToNext(SelectingSequence.java:60)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.hasNext(SelectingSequence.java:32)
	at jetbrains.mps.internal.collections.runtime.IterableUtils.first(IterableUtils.java:59)
	at jetbrains.mps.internal.collections.runtime.AbstractSequence.first(AbstractSequence.java:187)
	at de.q60.mps.polymorphicfunctions.runtime.DefaultPFContext.callFunction(DefaultPFContext.java:35)
	at de.q60.mps.polymorphicfunctions.runtime.ContextPFContext.callFunction(ContextPFContext.java:22)
	at de.q60.mps.polymorphicfunctions.runtime.AbstractPFContext.callFunction(AbstractPFContext.java:11)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$2$1.run(InteractionServer.java:190)
	at de.q60.mps.polymorphicfunctions.runtime.ContextValue.runWith(ContextValue.java:34)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4$2.contextCall(InteractionServer.java:188)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4.invoke(InteractionServer.java:194)
	at org.modelix.ui.sm.server.plugin.InteractionServer$4.invoke(InteractionServer.java:1)
	at org.modelix.ui.sm.server.plugin._Adapters$_return_P1_E0_to__void_P1_E0_adapter.invoke(_Adapters.java:14)
	at org.modelix.ui.common.UiStateUtil$9.invoke(UiStateUtil.java:166)
	at org.modelix.ui.common.UiStateUtil$9.invoke(UiStateUtil.java:1)
	at org.modelix.ui.common.UiStateUtil$4$1.invoke(UiStateUtil.java:76)
	at org.modelix.model.api.PBranch$computeWrite$1.invoke(PBranch.kt:76)
	at org.modelix.model.api.PBranch$computeWrite$1.invoke(PBranch.kt:20)
	at org.modelix.model.api.ContextValue.computeWith(ContextValue.kt:38)
	at org.modelix.model.api.PBranch.runWrite(PBranch.kt:53)
	at org.modelix.model.api.PBranch.computeWrite(PBranch.kt:76)
	at org.modelix.model.area.PArea$executeWrite$1.invoke(PArea.kt:60)
	at org.modelix.model.api.ContextValue.computeWith(ContextValue.kt:38)
	at org.modelix.model.area.ContextArea.offer(ContextArea.kt:35)
	at org.modelix.model.area.PArea.executeWrite(PArea.kt:60)
	at org.modelix.ui.common.UiStateUtil$4.select(UiStateUtil.java:74)
	at org.modelix.ui.common.UiStateUtil$4.select(UiStateUtil.java:1)
	at jetbrains.mps.internal.collections.runtime.ISelector.invoke(ISelector.java:12)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.moveToNext(SelectingSequence.java:60)
	at jetbrains.mps.internal.collections.runtime.impl.SelectingSequence$SelectingIterator.hasNext(SelectingSequence.java:32)
	at jetbrains.mps.internal.collections.runtime.ListSequence.fromIterable(ListSequence.java:122)
	at jetbrains.mps.internal.collections.runtime.AbstractSequence.toListSequence(AbstractSequence.java:215)
	at org.modelix.ui.common.UiStateUtil.writeOnStateRoots(UiStateUtil.java:80)
	at org.modelix.ui.common.UiStateUtil.writeOnSession(UiStateUtil.java:161)
	at org.modelix.ui.sm.server.plugin.InteractionServer.processMessage(InteractionServer.java:154)
	at org.modelix.ui.sm.server.plugin.InteractionServer$1$1$1.run(InteractionServer.java:93)
	at de.q60.mps.polymorphicfunctions.runtime.ContextValue.runWith(ContextValue.java:34)
	at org.modelix.ui.sm.server.plugin.InteractionServer$1$1.run(InteractionServer.java:91)
	at de.q60.mps.incremental.util.ContextValue.runWith(ContextValue.java:34)
	at org.modelix.ui.sm.server.plugin.InteractionServer$1.run(InteractionServer.java:89)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:216)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:24)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:199)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeAndWait$8(ApplicationImpl.java:476)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:126)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:501)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

--- a/mps/org.modelix.ui.sm/models/org.modelix.ui.sm.pf.mps
+++ b/mps/org.modelix.ui.sm/models/org.modelix.ui.sm.pf.mps
@@ -273,9 +273,6 @@
       </concept>
     </language>
     <language id="94b64715-a263-4c36-a138-8da14705ffa7" name="de.q60.mps.shadowmodels.transformation">
-      <concept id="7335687028107163797" name="de.q60.mps.shadowmodels.transformation.structure.TConceptType" flags="ig" index="02i3D">
-        <reference id="7335687028107163800" name="concept" index="02i3$" />
-      </concept>
       <concept id="7207189200502820768" name="de.q60.mps.shadowmodels.transformation.structure.OperationCall" flags="ng" index="p1pJ4">
         <reference id="7207189200502820817" name="operation" index="p1pIP" />
         <child id="7207189200503148686" name="parameterValues" index="p08jE" />
@@ -4981,8 +4978,8 @@
             <node concept="3cpWs6" id="fizU034X0c" role="3cqZAp">
               <node concept="2ShNRf" id="fizU034X0d" role="3cqZAk">
                 <node concept="kMnCb" id="fizU034X0e" role="2ShVmc">
-                  <node concept="02i3D" id="fizU034X0f" role="kMuH3">
-                    <ref role="02i3$" to="j481:1nlzgMD9WVV" resolve="IAction" />
+                  <node concept="3Tqbb2" id="DbjyVuS2pL" role="kMuH3">
+                    <ref role="ehGHo" to="j481:1nlzgMD9WVV" resolve="IAction" />
                   </node>
                 </node>
               </node>
@@ -5111,8 +5108,8 @@
             <node concept="3cpWs6" id="fizU034QAk" role="3cqZAp">
               <node concept="2ShNRf" id="fizU034Wya" role="3cqZAk">
                 <node concept="kMnCb" id="fizU034WZE" role="2ShVmc">
-                  <node concept="02i3D" id="fizU034WZO" role="kMuH3">
-                    <ref role="02i3$" to="j481:1nlzgMD9WVV" resolve="IAction" />
+                  <node concept="3Tqbb2" id="DbjyVuS2q4" role="kMuH3">
+                    <ref role="ehGHo" to="j481:1nlzgMD9WVV" resolve="IAction" />
                   </node>
                 </node>
               </node>

